x/inflation/types/v1beta2/inflation_calculator.go

```diff
@@ -0,0 +1,60 @@
+package v1beta2
+
+import (
+	"math"
+	"strconv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+func GetInflationCalculator(
+	genesisTime time.Time,
+	inflationParamSubspace paramstypes.Subspace,
+) minttypes.InflationCalculationFn {
+	return func(ctx sdk.Context, minter minttypes.Minter, params minttypes.Params, bondedRatio sdk.Dec) sdk.Dec {
+		var inflationParams Params
+		inflationParamSubspace.GetParamSet(ctx, &inflationParams)
+		tHalf := float64(inflationParams.InflationDecayFactor)
+
+		// Number of hours in an year = 8766 (averaging the leap year hours)
+		// Number of minutes in an hour = 60
+		// Number of seconds in a minute = 60
+		// => Number of seconds per year = 60 * 60 * 8766 = 31557600
+		t := ctx.BlockTime().Sub(genesisTime).Seconds() / 31557600 // years passed (can be a fraction, eg: 0.5)
+		idealInflation := inflationParams.InitialInflation.Mul(sdk.MustNewDecFromStr(
+			strconv.FormatFloat(math.Pow(2, -t/tHalf), 'f', -1, 64),
+		))
```

Comment:
Use `sdk.Dec` or standard `big.*`.

Reply:
There's no `Pow()` equivalent on `sdk.Dec` or `big.*` that accepts a fraction. That's why using this way of formatting the `Pow()` float value to a string and parsing a `sdk.Dec` from that.

Reply:
I see.  How annoying.

How about using [this](https://github.com/robpike/ivy/blob/master/value/power.go#L24) from [this](https://github.com/golang/go/issues/14102), or something similar?

Reply:
Done